src/Components/Blazor/WebAssembly.Authentication/src/Options/DefaultOidcProviderOptionsConfiguration.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    internal class DefaultOidcOptionsConfiguration : IPostConfigureOptions<RemoteAuthenticationOptions<OidcProviderOptions>>
+    {
+        private readonly NavigationManager _navigationManager;
+
+        public DefaultOidcOptionsConfiguration(NavigationManager navigationManager) => _navigationManager = navigationManager;
+
+        public void Configure(RemoteAuthenticationOptions<OidcProviderOptions> options)
+        {
+            options.UserOptions.AuthenticationType ??= options.ProviderOptions.ClientId;
+
+            var redirectUri = options.ProviderOptions.RedirectUri;
+            if (redirectUri == null || !Uri.TryCreate(redirectUri, UriKind.Absolute, out _))
+            {
+                redirectUri ??= "authentication/login-callback";
+                options.ProviderOptions.RedirectUri = _navigationManager
+                    .ToAbsoluteUri(redirectUri).AbsoluteUri;
```

Comment:
What's the benefit in making these things into absolute URIs? I'd expect them to work as-is as base-relative ones.